bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -631,6 +636,18 @@ private void updateDynamicChannels(Capabilities capabilities) {
         removeListProcessChannel(removeList, CHANNEL_SHADE_VANE,
                 capabilities.supportsTiltAnywhere() || capabilities.supportsTiltOnClosed());
 
+        boolean batteryChannelsRequired;
+        try {
+            String batteryKind = shade.batteryKind;
+            batteryChannelsRequired = batteryKind == null || Integer.parseInt(batteryKind) != HARDWIRED_POWER_SUPPLY;
+        } catch (NumberFormatException e) {
+            batteryChannelsRequired = true;
+        }
+
+        removeListProcessChannel(removeList, CHANNEL_SHADE_BATTERY_LEVEL, batteryChannelsRequired);
+        removeListProcessChannel(removeList, CHANNEL_SHADE_BATTERY_VOLTAGE, batteryChannelsRequired);
```

Comment:
The voltage channel should not be removed as it works just as well for fixed power supplies as it does for batteries, and it can be used for verifying that the power supply (with voltage drop for long cables) is correctly set up, and for monitoring the voltage even after initial installation. Personally I have items bound to all voltage channels, also for fixed power supply.

The same is actually true for the other two channels, although they are less useful and accurate. The only problem I see is that the user now need to configure the battery kind in the PowerView app, which was previously not needed. If you want to go ahead with this change, I would therefore suggest to clearly mention this requirement in the README.

Reply:
Yes I think this is a new feature in the PowerView App, and probably in the motor firmware. When I bought my shades, I was told that rechargeable batteries were not supported. But now apparently they are..

So I will modify the ReadMe to explain that. And I will retain the battery voltage channel; although it should probably be rephrased as power supply voltage..

Reply:
PS I think the default setting in the App is batteryKind:2 so configuring something else, and thus losing the battery channels, is an act of knowing..

Reply:
I agree, I actually woke up this morning realizing that probably default is not fixed power supply. I have no memory about the default though, so I hope you are right. Default could also be null? Anyway, I think all that needs to be explained, assuming that default is not hardwired, is a short text for the two channels that they are only present when power source is configured as battery.

Reply:
In old JSON captured from my hub two years ago the element had a string value `"batteryKind":"unassigned"`, whereas in new JSON captured from my hub today it has a numeric value `"batteryKind":2` .. and I did not expressly change anything in the app, so I assume these (both old and new JSON) were/are indeed the default values.